ui/components/src/TimeRangeSelector/AbsoluteTimePicker.tsx

```diff
@@ -0,0 +1,149 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useState } from 'react';
+import { Box, Stack, TextField, Divider } from '@mui/material';
+import { LocalizationProvider, StaticDateTimePicker } from '@mui/x-date-pickers';
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+import { format } from 'date-fns';
+import { AbsoluteTimeRange } from '@perses-dev/core';
+import { validateDateRange } from './utils';
+
+const DATE_TIME_FORMAT = 'yyyy-MM-dd HH:mm:ss';
+
+interface AbsoluteTimeFormProps {
+  initialTimeRange: AbsoluteTimeRange;
+  onChange: (timeRange: AbsoluteTimeRange) => void;
+}
+
+export const AbsoluteTimePicker = ({ initialTimeRange, onChange }: AbsoluteTimeFormProps) => {
+  const [timeRange, setTimeRange] = useState<AbsoluteTimeRange>(initialTimeRange);
+  const [showStartCalendar, setShowStartCalendar] = useState<boolean>(true);
+  const [showEndCalendar, setShowEndCalendar] = useState<boolean>(false);
+
+  const handleOnStartAccept = () => {
+    setShowStartCalendar(false);
+    setShowEndCalendar(true);
+  };
+
+  const handleOnEndAccept = () => {
+    setShowEndCalendar(false);
+    setShowStartCalendar(true);
+    onChange(timeRange);
+  };
+
+  return (
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
+      <Stack spacing={3} sx={{ padding: (theme) => theme.spacing(1, 2, 3) }}>
+        {showStartCalendar && (
+          <Box
+            sx={(theme) => ({
+              // TODO: create separate reusable calendar component
+              '.MuiPickerStaticWrapper-content': {
+                backgroundColor: theme.palette.background.default,
+              },
+              // reposition AM and PM buttons
+              '.MuiIconButton-sizeMedium': {
+                top: 80,
+                bottom: 'auto',
+                margin: theme.spacing(0, 3),
+              },
+            })}
+          >
+            <h3>Select Start Time</h3>
+            <StaticDateTimePicker
+              displayStaticWrapperAs="desktop"
+              openTo="day"
+              disableHighlightToday={true}
+              value={initialTimeRange.start}
+              onChange={(newValue) => {
+                if (newValue === null) return;
+                setTimeRange({ start: newValue, end: timeRange.end });
+              }}
+              onAccept={() => handleOnStartAccept()}
+              renderInput={(params) => <TextField {...params} />}
+            />
+          </Box>
+        )}
+        {showEndCalendar && (
```

Comment:
Do we need showEndCalendar? Can we just do `!showStartCalendar`?

Reply:
Good call! I changed this so only showStartCalendar is used now